File: mycollab-mobile/src/main/scala/com/mycollab/mobile/module/project/view/parameters/RiskScreenData.scala
/**
 * This file is part of mycollab-mobile.
 *
 * mycollab-mobile is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-mobile is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-mobile.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.mobile.module.project.view.parameters

import com.mycollab.module.project.domain.Risk
import com.mycollab.module.project.domain.criteria.RiskSearchCriteria
import com.mycollab.vaadin.mvp.ScreenData

/**
  * @author MyCollab Ltd
  * @since 5.2.7
  */
object RiskScreenData {

  class Search(param: RiskSearchCriteria) extends ScreenData(param) {}

  class Read(param: Integer) extends ScreenData(param) {}

  class Add(param: Risk) extends ScreenData(param) {}

  class Edit(param: Risk) extends ScreenData(param) {}

}

File: mycollab-web/src/main/java/com/mycollab/module/project/view/parameters/RiskScreenData.kt
package com.mycollab.module.project.view.parameters

import com.mycollab.module.project.domain.Risk
import com.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object RiskScreenData {
    class Read(params: Int) : ScreenData<Int>(params)

    class Add(param: Risk) : ScreenData<Risk>(param)

    class Edit(param: Risk) : ScreenData<Risk>(param)
}